Sources/FoundationEssentials/FileManager/FileManager+Files.swift

```diff
@@ -362,32 +378,86 @@ extension _FileManagerImpl {
         isDirectory = result.isDirectory
         return true
     }
-    
+
+#if !os(Windows)
     private func _fileAccessibleForMode(_ path: String, _ mode: Int32) -> Bool {
         path.withFileSystemRepresentation { ptr in
             guard let ptr else { return false }
             return access(ptr, mode) == 0
         }
     }
-    
+#endif
+
     func isReadableFile(atPath path: String) -> Bool {
+#if os(Windows)
+        do {
+            return try path.withNTPathRepresentation {
+                var faAttributes: WIN32_FILE_ATTRIBUTE_DATA = .init()
+                return GetFileAttributesExW($0, GetFileExInfoStandard, &faAttributes) == true
```

Comment:
```suggestion
                return GetFileAttributesExW($0, GetFileExInfoStandard, &faAttributes)
```

Does that work, or does this return an optional?